### DISCUSSION TITLE
is utils.k_hop_subgraph() method can be processed parallelly?

### DISCUSSION THREAD

**Discussion by ybkim-dev** [pyg-team/pytorch_geometric#2536] [Q&A]
is utils.k_hop_subgraph() method can be processed parallelly?

I want to know whether it is true or false

**Comment by rusty1s** [comment]
It is parallelized across the node dimension, but there is no way to parallelize it across hops since the input of each depends on the output of the previous call.

**Reply by ybkim-dev** [reply]
okay, thank you for reply!

well, i have one more specific question to understand your answer.

so you mean for example, node1 and node2's subgraph can be made in different machine but, node 1's subgraph should be made in same machine and also, node 2's  too.

is this right?

**Reply by rusty1s** [reply]
Are you referring to distributed parallelization? That's not possible without a more sophisticated implementation. Otherwise, you are correct.

Here, I was just referring to vectorization (e.g., standard PyTorch/numpy functionality) as parallelization technique. For example, for each node we can gather the local 1-hop neighborhood and merge them together in parallel. However, we need to do that sequentially for each iteration 1 to `k`.

**Reply by ybkim-dev** [reply]
Thank you a lot!

you give me a perfect answer!
